services/alerts/src/useAlert.ts

```diff
@@ -1,27 +1,51 @@
-import { useContext } from 'react'
+import { useContext, useRef, useCallback, useMemo } from 'react'
 import { AlertsManagerContext } from './AlertsManagerContext'
-import { Alert, AlertOptions } from './types'
+import {
+    AlertBarComponent,
+    AlertOptions,
+    AlertsManager,
+    AlertsManagerAlert,
+} from './types'
 
 export const useAlert = (
     message: string | ((props: any) => string),
     options: AlertOptions | ((props: any) => AlertOptions) = {}
 ) => {
-    const alertsManager = useContext(AlertsManagerContext)
+    const alertsManager: AlertsManager = useContext(AlertsManagerContext)
+    const ref = useRef<AlertBarComponent>()
+    const alert: AlertsManagerAlert = useMemo(() => alertsManager.add(ref), [
```

Comment:
> You may rely on useMemo as a performance optimization, not as a semantic guarantee. In the future, React may choose to “forget” some previously memoized values and recalculate them on next render, e.g. to free memory for offscreen components. Write your code so that it still works without useMemo — and then add it to optimize performance.

(source: [React docs](https://reactjs.org/docs/hooks-reference.html#usememo))

Here we're assuming that `alertsManager.add` will only be called once, but it might be called again later which would be problematic - should be `useRef` I think

Reply:
Or better yet, create the alert in `show`